lucene/core/src/java/org/apache/lucene/index/MergePolicy.java

```diff
@@ -399,6 +427,23 @@ public String segString(Directory dir) {
       }
       return b.toString();
     }
+
+    /**
+     * Waits if necessary for at most the given time for all merges.
+     */
+    boolean await(long timeout, TimeUnit unit) {
+      try {
+        CompletableFuture<Void> future = CompletableFuture.allOf(merges.stream()
+            .map(m -> m.completable).collect(Collectors.toList()).toArray(new CompletableFuture<?>[0]));
+        future.get(timeout, unit);
+        return true;
+      } catch (InterruptedException e) {
+        Thread.interrupted();
```

Comment:
In my change, I remember I originally tried handling the `InterruptedException` while waiting for the `CountDownLatch` in a similar way, but it caused an intermittent test failure (in `TestIndexWriterWithThreads`, maybe?) You might run into the same test failure once you start calling this `await` method from within `IndexWriter`.

I was able to get the test passing by rethrowing it wrapped in a `ThreadInterruptedException`.

Reply:
done